cli/cmd/upgrade.go

```diff
@@ -265,15 +265,6 @@ func upgrade(ctx context.Context, k *k8s.KubernetesAPI, flags []flag.Flag, stage
 	if err != nil {
 		return bytes.Buffer{}, fmt.Errorf("failed to load stored values: %w", err)
 	}
-	// If there is no linkerd-config-overrides secret, assume we are upgrading
-	// from a version of Linkerd prior to the introduction of this secret.  In
-	// this case we load the values from the legacy linkerd-config configmap.
-	if values == nil {
-		values, err = loadStoredValuesLegacy(ctx, k)
-		if err != nil {
-			return bytes.Buffer{}, err
-		}
-	}
 
 	// If values is still nil, then neither the linkerd-config-overrides secret
```

Comment:
Good catch! Updated in https://github.com/linkerd/linkerd2/pull/7309/commits/212591032d05f2deb28502d68b5297ec4619150b

Reply:
@krzysztofdrys Sorry for not making it clear. I was mentioning about the comment as it mentions `neither the linkerd-config-overrides secret nor the legacy values were found` which is not the case now that legacy values logic is removed right? Can we make the comment to mention only about the secret to prevent confusion? 🤔

Reply:
I think you made yourself pretty clear, it's just that I pushed my change into a wrong branch and put a wrong link into a comment. 🙃 

I hope this is what you meant: https://github.com/linkerd/linkerd2/commit/1d90a802f66271d19faee14e48e0c534b5a964ae

Reply:
Ahhhh, Makes sense. Now I can see the change here.